File: fixtures/ruby/lib/math_utils.rb
# frozen_string_literal: true

# Simple math utility module for testing.
module MathUtils
  def self.add(a, b)
    a + b
  end

  def self.multiply(a, b)
    a * b
  end

  def self.divide(a, b)
    raise ArgumentError, 'Division by zero' if b == 0

    a / b
  end

  def self.fibonacci(n)
    raise ArgumentError, 'Negative index not allowed' if n < 0
    return 0 if n == 0
    return 1 if n == 1

    fibonacci(n - 1) + fibonacci(n - 2)
  end

  def self.palindrome?(str)
    cleaned = str.downcase.gsub(/[^a-z0-9]/, '')
    cleaned == cleaned.reverse
  end
end

File: fixtures/python/src/math_utils.py
"""
Simple math utility functions for testing.
"""


def add(a: int | float, b: int | float) -> int | float:
    """Add two numbers."""
    return a + b


def multiply(a: int | float, b: int | float) -> int | float:
    """Multiply two numbers."""
    return a * b


def divide(a: int | float, b: int | float) -> int | float:
    """Divide two numbers."""
    if b == 0:
        raise ValueError("Division by zero")
    return a / b


def fibonacci(n: int) -> int:
    """Return the nth Fibonacci number."""
    if n < 0:
        raise ValueError("Negative index not allowed")
    if n == 0:
        return 0
    if n == 1:
        return 1
    return fibonacci(n - 1) + fibonacci(n - 2)


def is_palindrome(s: str) -> bool:
    """Check if a string is a palindrome."""
    cleaned = "".join(c.lower() for c in s if c.isalnum())
    return cleaned == cleaned[::-1]
